File: backend/src/main/java/com/lifeos/goal/service/GoalService.java
package com.lifeos.goal.service;

import com.lifeos.goal.dto.CreateGoalRequest;
import com.lifeos.goal.dto.GoalDetailResponse;
import com.lifeos.goal.dto.GoalResponse;
import com.lifeos.goal.dto.UpdateGoalRequest;
import com.lifeos.goal.entity.GoalStatus;

import java.util.List;

public interface GoalService {

    GoalResponse createGoal(Long userId, CreateGoalRequest request);

    List<GoalResponse> getGoals(Long userId, GoalStatus status);

    GoalDetailResponse getGoalById(Long userId, Long goalId);

    GoalResponse updateGoal(Long userId, Long goalId, UpdateGoalRequest request);

    void deleteGoal(Long userId, Long goalId);

    void recalculateProgress(Long goalId, Long userId);
}

File: frontend/src/services/goalService.js
import api from './api';

export const goalService = {
  async getGoals(status) {
    const params = status ? { status } : {};
    const response = await api.get('/goals', { params });
    return response.data;
  },

  async getGoalById(id) {
    const response = await api.get(`/goals/${id}`);
    return response.data;
  },

  async createGoal(payload) {
    const response = await api.post('/goals', payload);
    return response.data;
  },

  async updateGoal(id, payload) {
    const response = await api.put(`/goals/${id}`, payload);
    return response.data;
  },

  async deleteGoal(id) {
    const response = await api.delete(`/goals/${id}`);
    return response.data;
  },
};

export default goalService;
